stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/CSS.java

```diff
@@ -136,57 +113,46 @@ public void run() {
             elementForNodeId,
             new StyleAccumulator() {
               @Override
-              public void store(String name, String value, boolean isDefault) {
+              public void store(String selector, String name, String value, boolean isDefault) {
                 if (!isDefault) {
-                  CSSProperty property = new CSSProperty();
-                  property.name = name;
-                  property.value = value;
-
-                  match.rule.style.cssProperties.add(property);
-                }
-              }
-            });
+                  List<CSSProperty> properties = selectors.get(selector);
+                  if (properties == null) {
+                    properties = new ArrayList<>();
+                    selectors.put(selector, properties);
+                  }
 
-        mDocument.getElementAccessibilityStyles(
-            elementForNodeId,
-            new StyleAccumulator() {
-              @Override
-              public void store(String name, String value, boolean isDefault) {
-                if (!isDefault) {
-                  CSSProperty property = new CSSProperty();
+                  final CSSProperty property = new CSSProperty();
                   property.name = name;
                   property.value = value;
-
-                  accessibilityMatch.rule.style.cssProperties.add(property);
+                  properties.add(property);
                 }
               }
             });
       }
     });
 
-    result.inherited = Collections.emptyList();
-    result.pseudoElements = Collections.emptyList();
+    final GetMatchedStylesForNodeResult result = new GetMatchedStylesForNodeResult();
+    result.matchedCSSRules = new ArrayList<>();
 
-    return result;
-  }
+    for (String selectorName : selectors.keySet()) {
```

Comment:
Here's another reason i'd prefer to avoid `HashMap` ... iterating over this requires a temporary `Iterator` allocation. Preferable to avoid, although not dogmatically so.